src/scenario.c

```diff
@@ -129,7 +130,16 @@ int scenario_load_and_play_from_path(const char *path)
 {
 	window_close_construction_windows();
 
-	if (!scenario_load(path))
+	uint32 extension = get_file_extension_type(path);
+	if (extension == FILE_EXTENSION_SC6) {
+		if (!scenario_load(path))
+			return 0;
+	}
+	else if (extension == FILE_EXTENSION_SC4) {
+		if (!rct1_load_scenario(path))
+			return 0;
+	}
+	else
 		return 0;
```

Comment:
Could you put braces around this line? It's more consistent with the preceding if and else-if clause.